dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelper.java

```diff
@@ -0,0 +1,331 @@
+
+        private static final ESMappingUtilHelper INSTANCE = new ESMappingUtilHelper();
+    }
+
+    public static ESMappingUtilHelper getInstance() {
+        return ESMappingUtilHelper.SingletonHolder.INSTANCE;
+    }
+
+    private ESMappingUtilHelper() {
+        contentTypeAPI = APILocator.getContentTypeAPI(APILocator.systemUser());
+        esMappingAPI = new ESMappingAPIImpl();
+        relationshipAPI = APILocator.getRelationshipAPI();
+    }
+
+    @VisibleForTesting
+    ESMappingUtilHelper(final ContentTypeAPI contentTypeAPI, final ESMappingAPIImpl esMappingAPI,
+            final RelationshipAPI relationshipAPI) {
+        this.contentTypeAPI = contentTypeAPI;
+        this.esMappingAPI = esMappingAPI;
+        this.relationshipAPI = relationshipAPI;
+    }
+
+    /**
+     * Sets a custom index mapping for relationships and also for mapping defined on field variables
+     * using `esCustomMapping` property
+     *
+     * @param indexName - Index where mapping will be updated
+     */
+    @CloseDBIfOpened
+    public void addCustomMapping(final String indexName) {
+
+        final Set<String> mappedFields = addCustomMappingFromFieldVariables(indexName);
+
+        addCustomMappingForRelationships(indexName, mappedFields);
+
+        addMappingForRemainingFields(indexName, mappedFields);
+    }
+
+    /**
+     * Sets a mapping for all relationships except for those that contains its custom mapping using
+     * field variables
+     *
+     * @param indexName - Index where mapping will be updated
+     * @param mappedFields - Collection that contains the fields with a specific mapping until now.
+     * </br> When a put mapping request is sent to Elasticsearch for each relationship (if needed),
+     * a new entry is added to the <b>mappedFields</b> collection
+     */
+    private void addCustomMappingForRelationships(final String indexName,
+            final Set<String> mappedFields) {
+        final List<Relationship> relationships = relationshipAPI.dbAll();
+
+        for (final Relationship relationship : relationships) {
+            final String relationshipName = relationship.getRelationTypeValue().toLowerCase();
+            if (!mappedFields.contains(relationshipName)) {
+                final JSONObject properties = new JSONObject();
+                try {
+                    properties.put("properties", new JSONObject()
+                            .put(relationshipName,
+                                    new JSONObject("{\n"
+                                            + "\"type\":  \"keyword\",\n"
+                                            + "\"ignore_above\": 8191\n"
+                                            + "}")));
+                    esMappingAPI.putMapping(indexName, properties.toString());
+
+                    //Adds to the set the mapped already set for this field
+                    mappedFields.add(relationshipName);
+                } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5739222)